ivy/stateful/optimizers.py

```diff
@@ -180,13 +188,11 @@ def __init__(
         compile_on_next_step
             Whether to compile the optimizer on the next step. Default is False.
         """
-        Optimizer.__init__(
-            self, lr, inplace, stop_gradients, compile_on_next_step=compile_on_next_step
-        )
+        Optimizer.__init__(self, lr, inplace, stop_gradients, compile_on_next_step)
```

Comment:
The usage of the kword argument (compile_on_next_step=compile_on_next_step) is necessary because compile_on_next is the 6th argument and it is being passed as the 5th argument here.